receiver/windowsservicereceiver/scraper.go

```diff
@@ -1,48 +1,145 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
-
-//revive:disable:unused-parameter
 //go:build windows
 
-package windowsservicereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsservicereceiver"
+package windowsservicereceiver
 
 import (
 	"context"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsservicereceiver/internal/metadata"
 )
 
-//nolint:unused
 type windowsServiceScraper struct {
-	scm      serviceManager
-	settings receiver.Settings
-	conf     *Config
-	mb       *metadata.MetricsBuilder
+	logger     *zap.Logger
+	cfg        *Config
+	mb         *metadata.MetricsBuilder
+	mgr        serviceManager
+	includeSet map[string]struct{}
+	excludeSet map[string]struct{}
+}
+
+func newWindowsServiceScraper(settings receiver.Settings, cfg *Config, mb *metadata.MetricsBuilder) *windowsServiceScraper {
+	ws := &windowsServiceScraper{
+		logger: settings.Logger,
+		cfg:    cfg,
+		mb:     mb,
+		mgr:    serviceManager{},
+	}
+	if len(cfg.IncludeServices) > 0 {
+		ws.includeSet = make(map[string]struct{}, len(cfg.IncludeServices))
+		for _, n := range cfg.IncludeServices {
+			ws.includeSet[n] = struct{}{}
+		}
+	}
+	if len(cfg.ExcludeServices) > 0 {
+		ws.excludeSet = make(map[string]struct{}, len(cfg.ExcludeServices))
+		for _, n := range cfg.ExcludeServices {
+			ws.excludeSet[n] = struct{}{}
+		}
+	}
+	return ws
 }
 
-//nolint:unused
-func newWindowsServiceScraper(settings receiver.Settings, _ *Config) windowsServiceScraper {
-	return windowsServiceScraper{
-		settings: settings,
-		mb:       metadata.NewMetricsBuilder(metadata.DefaultMetricsBuilderConfig(), settings),
+func mapStateToMetricValue(st State) int {
+	switch st {
+	case StateStopped:
+		return 1
+	case StateStartPending:
+		return 2
+	case StateStopPending:
+		return 3
+	case StateRunning:
+		return 4
+	case StateContinuePending:
+		return 5
+	case StatePausePending:
+		return 6
+	case StatePaused:
+		return 7
+	default:
+		return 1
 	}
 }
 
-//nolint:unused
-func (*windowsServiceScraper) start(context.Context, component.Host) (err error) {
-	return nil
+func mapStartTypeToAttr(st StartType, _ bool) metadata.AttributeStartupMode {
```

Comment:
Since it is a private function the addition of the not used `bool` can be delayed until it is really needed.